algoliasearch-client-js/algoliasearch-client-js.d.ts

```diff
@@ -0,0 +1,1563 @@
+     */
+    initIndex(name:string):IAlgoliaIndex;
+    /**
+     * Query on multiple index
+     * @param queries index name, query and query parameters
+     * @param cb callback(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#multiple-queries---multiplequeries
+     */
+    search(queries:{ indexName:string, query:string, options:IAlgoliaQueryParameters }, cb:Function) : void;
+    /**
+     * Query on multiple index
+     * @param queries index name, query and query parameters
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#multiple-queries---multiplequeries
+     */
+    search(queries:{ indexName:string, query:string, options:IAlgoliaQueryParameters }) : Promise<IAlgoliaResponse>;
+    /**
+     * clear browser cache
+     * https://github.com/algolia/algoliasearch-client-js#cache
+     */
+    clearCache() : void;
+    /**
+     * kill alive connections
+     * https://github.com/algolia/algoliasearch-client-js#keep-alive
+     */
+    destroy() : void;
+    /**
+     * List all your indices along with their associated information (number of entries, disk size, etc.)
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#list-indices---listindexes
+     */
+    listIndexes(cb:Function) : void;
+    /**
+     * List all your indices along with their associated information (number of entries, disk size, etc.)
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#list-indices---listindexes
+     */
+    listIndexes(): Promise<any>;
+    /**
+     * Delete a specific index
+     * @param name
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#delete-index---deleteindex
+     */
+    deleteIndex(name:string, cb:Function) : void;
+    /**
+     * Delete a specific index
+     * @param name
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#delete-index---deleteindex
+     */
+    deleteIndex(name:string) : Promise<any>;
+    /**
+     * Copy an  index from a specific index to a new one
+     * @param from origin index
+     * @param to destination index
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#copy-index---copyindex
+     */
+    copyIndex(from:string, to:string, cb:Function) : void;
+    /**
+     * Copy an  index from a specific index to a new one
+     * @param from origin index
+     * @param to destination index
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#copy-index---copyindex
+     */
+    copyIndex(from:string, to:string) : Promise<any>;
+    /**
+     * Move index to a new one (and will overwrite the original one)
+     * @param from origin index
+     * @param to destination index
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#move-index---moveindex
+     */
+    moveIndex(from:string, to:string, cb:Function) : void;
```

Comment:
Provide a more specific type for the callback -- looks like it takes two parameters, the first an Error and the other some result type (`any` if you can't get anything better).